homeassistant/components/scrape/config_flow.py

```diff
@@ -184,7 +216,17 @@ async def validate_sensor_edit(
     # Standard behavior is to merge the result with the options.
     # In this case, we want to add a sub-item so we update the options directly.
     idx: int = handler.flow_state["_idx"]
-    handler.options[SENSOR_DOMAIN][idx].update(user_input)
+    sensor_options = handler.options[SENSOR_DOMAIN][idx]
+
+    __validate_enum_state(
+        CONF_DEVICE_CLASS, SensorDeviceClass, user_input, sensor_options
+    )
+    __validate_enum_state(
+        CONF_STATE_CLASS, SensorStateClass, user_input, sensor_options
+    )
+    __validate_enum_state(CONF_UNIT_OF_MEASUREMENT, None, user_input, sensor_options)
+
+    sensor_options.update(user_input)
```

Comment:
```suggestion
    handler.options[SENSOR_DOMAIN][idx].update(user_input)
```

Reply:
The key is the adjusted suggestion above: simply handling when the key is missing from the user-input, and removing the corresponding item manually.